diff --git a/inst/include/boost/simd/swar/functions/details/topology.hpp b/inst/include/boost/simd/swar/functions/details/topology.hpp
index ceba192..941fcf4 100644
--- a/inst/include/boost/simd/swar/functions/details/topology.hpp
+++ b/inst/include/boost/simd/swar/functions/details/topology.hpp
@@ -126,7 +126,7 @@ namespace boost { namespace simd { namespace details
   #define M3(z,n,t)                                                            \
   template<typename Permutation> struct topology<Permutation,n>                \
   {                                                                            \
-    BOOST_PP_REPEAT(n,M0,n);                                                   \
+    BOOST_PP_REPEAT(n,M0,n)                                                    \
                                                                                \
     typedef boost::mpl::                                                       \
             int_< true BOOST_PP_REPEAT(n,M1,n)                                 \
